src/YoutubeDl.php

```diff
@@ -111,7 +111,11 @@ public function onProgress(callable $onProgress)
 
     public function download(string $url): Video
     {
-        if (!$this->downloadPath) {
+        if (isset($this->options['output']) && is_dir(dirname($this->options['output'])) && '.' != dirname($this->options['output'])) {
+            $this->downloadPath = dirname($this->options['output']);
```

Comment:
Please add a local variable for this, we don't want side effects here, because downloadPath would be changed on each download call :)

Reply:
Hi,

Finally I have removed all logic related with the new option, but I have keep some improvements related with the output parser in processDownload function.

With this changes, if I specify the output directory equals to download path it works as expected, so I prefer dont make changes ir your logic (keeping it simple) in this way I can adapt in my code without problems.

The improvement is the same which I talked at the start of the thread, we only need the filename for generate the path for decode the content from processDownload but If user specified a output template with a full path, the output print it was not valid:

Writing video description metadata as JSON to: C:\shared\Phonebloks.info.json
vs
Writing video description metadata as JSON to: Phonebloks.info.json

Thank you.